tests/testthat/reporters/tests.R

```diff
@@ -33,7 +33,7 @@ test_that("Error:3", {
   g <- function() {h()}
   h <- function() {stop("!")}
 
-  f()
+  expect_message(f(), NA)
```

Comment:
Ooooh, because the problem isn't catching the error message, it's that the `end_test()` method gets called inside the sink.